mpqp/core/instruction/gates/native_gates.py

```diff
@@ -868,6 +946,40 @@ def qiskit_gate(cls):
 
         return UGate
 
+    @classproperty
+    def cirq_gate(cls):
+        from cirq.circuits.qasm_output import QasmUGate
+        from cirq.ops.common_gates import ry as cirq_ry, rz as cirq_rz
+        from cirq.ops.global_phase_op import GlobalPhaseGate
+        from cirq.ops.raw_types import Qid
+
+        class CirqUGate(QasmUGate):  # pyright: ignore[reportUntypedBaseClass]
+            def __init__(
+                self, theta, phi, lmda  # pyright: ignore[reportMissingParameterType]
+            ) -> None:
+                self.lmda = lmda
+                self.theta = theta
+                self.phi = phi
+
+            def __repr__(self) -> str:
+                return (
+                    f'U('
+                    f'theta={self.theta !r}, '
```

Comment:
Why `!r` is used for theta and hi, and not for lambda ?

Another question concerning those angles, no need to tweak them with some division by pi ? When you can ry and rz in the decompose ?